### PR TITLE
gh-pages: more redirect fixes

### LINKED ISSUE #27904: docs: redirect root pages to the latest version
Like `/about/` to `4.2/about/` etc.

Here's the list

```
about
browser-bugs
components
css
customize
examples
    blog
    carousel
    cover
    dashboard
    grid
    jumbotron
    jumbotron-narrow
    justified-nav
    navbar
    navbar-fixed-top
    navbar-static-top
    non-responsive
    offcanvas
    signin
    starter-template
    sticky-footer
    sticky-footer-navbar
    theme
    tooltip-viewport
getting-started
javascript
migration -> to /docs/3.4/migration/ ?
```

### PR DESCRIPTION
**Pull request by XhmikosR**
Fixes #27904

### CODE REVIEW

**Comment by MartijnCuppens** [comment]
Can we make use of a redirect template instead of all these similar files?

**Comment by XhmikosR** [comment]
There is a template when possible in the main branches. These are direct fixes.

File: docs/4.0/components/index.html [hunk 1]
```diff
@@ -1,38 +1,11 @@
...
-        text-decoration: none;
-      }
-    </style>
-  </head>
-  <body>
-    <h1>Redirecting&hellip;</h1>
-    <a href="https://getbootstrap.com/docs/4.0/components/buttons/">Click here if you are not redirected</a>
-    <script>window.location="https://getbootstrap.com/docs/4.0/components/buttons/";</script>
-  </body>
+<!DOCTYPE html>
```
Comment (mdo) [inline docs/4.0/components/index.html]: No need to capitalize `doctype`.

File: docs/4.0/components/index.html [hunk 2]
```diff
@@ -1,38 +1,11 @@
...
+<html lang="en-US">
+  <meta charset="utf-8">
+  <title>Redirecting&hellip;</title>
+  <link rel="canonical" href="https://getbootstrap.com/docs/4.0/components/buttons/">
```
Comment (mdo) [inline docs/4.0/components/index.html]: Do we need to move these `canonical` links to point to 4.2? I recall thinking about these as something to do with how we version our docs.

**Comment by XhmikosR** [comment]
This is the default template which is using uppercase doc type.

**Comment by XhmikosR** [comment]
The canonical is set to the redirect target.

**Review by mdo — APPROVED** [review]

**Pull request merged**